server/src/main/java/org/elasticsearch/node/NodeService.java

```diff
@@ -133,6 +140,20 @@ public NodeInfo info(
         );
     }
 
+    private static Map<String, Integer> findComponentVersions() {
+        var versions = ServiceLoader.load(ComponentVersionNumber.class)
```

Comment:
I'm not sure this is going to work as is.  I believe this will only find SPI impls in the system classloader (it's actually the thread context classloader, but that should be the systemclassloader here). In order to allow finding impls from plugins, we need to use `PluginsService.loadServiceProviders`

Reply:
It loads it properly in the xpack module - but thats maybe because that is in the system classloader when running tests there. I'll take another look.

Reply:
Ah yes, it needs the plugins loader to actually get the right classes. Updated.